### PR TITLE
Document addons

### PR DESCRIPTION
**Pull request by parisk**
Closes sourcelair/xterm.js#306.

### CODE REVIEW

File: _docs/api/Terminal.md [hunk 1]
```diff
@@ -126,6 +126,25 @@ Retrieves an option from the terminal.
 var doesTheTerminalCursorBlink = term.getOption('cursorBlink');
 ```
 
+### `loadAddon(addon[, callback])`
+
+- `addon` - String - The name of the add-on to load
+- `callback` - Function - The callback to run after running the add-on ()
+
+Attempts to load an add-on using [CommonJS](https://nodejs.org/docs/latest/api/modules.html) or with [RequireJS](http://requirejs.org/), if [CommonJS](https://nodejs.org/docs/latest/api/modules.html) is not available. If none of them is available a [console error](https://developer.mozilla.org/es/docs/Web/API/Console/error) is being printed.
```
Comment (Tyriar) [inline _docs/api/Terminal.md]: Grammar improvement:

If none of them is available a [console error](https://developer.mozilla.org/es/docs/Web/API/Console/error) is being printed

->

If none of them are available, an error is sent to the [console](https://developer.mozilla.org/es/docs/Web/API/Console/error).

File: _docs/api/Terminal.md [hunk 2]
```diff
@@ -126,6 +126,25 @@ Retrieves an option from the terminal.
...
+
+> **Attention!** This is a static method, since it extends the basic `Terminal` prototype and not a single `Terminal` instance.
+
+```javascript
+Terminal.loadAddon('fit'); // Loads the `fit` addon and adds it's methods to the Terminal prototype
```
Comment (Tyriar) [inline _docs/api/Terminal.md]: it's -> its
Terminal -> `Terminal`

File: _docs/api/addons/attach.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+---
+title: attach
+category: addon
+---
+
+The attach addon provides methods for attaching a terminal to a WebSocket stream, such as [Docker's WebSocket attach endpoint](https://docs.docker.com/engine/reference/api/docker_remote_api_v1.24/#/attach-to-a-container-websocket).
+
+This allows easy hooking of the terminal front-end to background processes and interact with it, just like you would do with in your local terminal. This means that the front-end terminal (a [`Terminal`](/docs/api/Terminal/) instance) will render the stdout and stderr logs of the back-end process and will send to it all keyboard and mouse events captured.
```
Comment (Tyriar) [inline _docs/api/addons/attach.md]: just like you would do with in your local terminal
->
just like you would do with a local terminal

File: _docs/api/addons/fullscreen.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+term.toggleFullscreen();  // Now the terminal should occupy the full viewport
+term.toggleFullscreen();  // Now the terminal should be back to normal
+```
+
+## Methods
+
+### `toggleFullscreen([fullscreen])`
+
+- `fullscreen` - Boolean - Whether to set terminal to fullscreen mode or not
```
Comment (Tyriar) [inline _docs/api/addons/fullscreen.md]: terminal -> the terminal

**Comment by parisk** [comment]
Thanks @Tyriar. I fixed all grammar errors in 41a0e2ec7a6fe81047a35767e2ad7bb5a2c3f39d.

**Pull request merged**